interview-questions.md

```diff
@@ -8,48 +8,49 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 3. What is an anonymous function in JavaScript?
 
-Your answer:
+Your answer: An anonymous function in JavaScript is a function that does not have a name. I believe it lives inside of a component or another function. A function that runs dependent on another function. I know it doesn't have or need a name. 
 
-Researched answer:
+Researched answer:Another definition of an annonymous function is a function that is stored in a variable and does not have or need a name. They are started by calling on the variables name. THey end with a semicolon and it can have many argunments but only one return.
 
 4. What is the difference between a class and an object?
 
-Your answer:
+Your answer: A class is a blue print for objects. Objects are the key value pairs that hold data, like arrays. Objects are the intersection of data and actions. A good real life example to illustrate this is classes are like recipes and objects are what we can create with recipes. 
 
-Researched answer:
+Researched answer: A class is just a blueprint for an object. For example if you have a class of Dogs we know that all dogs have a certain breed, color, size, age. Now if we look at a Boston Terrier dog for example the breed is Boston Terrier the size is small the age is xx so that dog itself is an object in the class of dogs which has its own attributes that are different from other dogs. .
 
 5. What is the difference between state and props in React?
 
-Your answer:
+Your answer: State in react is something that is Set in the App.js and allows us to pass down information to components via props. Props cannot pass data back to the parent or state. State can call upon the value index or array of an object. 
 
-Researched answer:
+Researched answer:State is used to manage data by using the setState() inside the parenthesis is where we pass the state we want to change. It can be used in classes and functions. States data is only available to the component. Props uses this data to pass down to components. 
 
 6. STRETCH: Which is the difference between a div and a span?
 
-Your answer:
+Your answer: A div is a blockline element and a span is inline. Span will only take up one line where as a div will consume the entire block. While we don't use it often a <span> element can provide space where needed in an HTML doc.
 
-Researched answer:
+Researched answer:DIVs aka division elements. DIVS take up the whole block and the span only takes up the part of the line selected. You can have more than one span on a line whereas you can only have one DIV per line or block . The divs come in handy when grouping similar paragraphs. SPan can be sued to style inline elements without affecting the rest of the code block. 
 
 ## Looking Ahead: Terms for Next Week
 
```

Comment:
Ruby can be a bit friendlier the javascript but I think it is because you have been exposed to programming by the time we introduce it.